adobe/name.hpp

```diff
@@ -63,12 +63,66 @@ constexpr std::size_t name_fnv_basis_k = sizesz_k == 8
                                              ? static_cast<std::size_t>(0xcbf29ce484222325ULL)
                                              : static_cast<std::size_t>(0x811c9dc5);
 
+// The following few items are helpers to multiply_size_t, which will multiply two size_t
+// values in a way that is guaranteed not to overflow. Otherwise, in Visual Studio 2017 we get
+// Warning C4307: '*': integral constant overflow
+
+constexpr std::size_t number_of_one_bits(std::size_t x)
+	{
+	return x == 0 ? 0 : 1 + number_of_one_bits(x & (x - 1));
+	}
+
+constexpr std::size_t size_t_bits_k = number_of_one_bits(~std::size_t(0));
+
+constexpr std::size_t size_t_lower(std::size_t x)
+	{
+	return x & (~(~std::size_t(0) << size_t_bits_k / 2));
+	}
+
+constexpr std::size_t size_t_upper(std::size_t x)
+	{
+	return size_t_lower(x >> size_t_bits_k / 2);
+	}
+
+// This multiplies two size_t and throws away the overflow in a way
+// that avoids overflow complaints from Visual Studio 2017:
+// Warning C4307: '*': integral constant overflow
+//
+constexpr std::size_t multiply_size_t(std::size_t x, std::size_t y)
+	{
```

Comment:
This is a lot of code and complexity to silence a warning for defined behavior. Please just suppress the warning locally.